src/templates/mdc-highlighter.ts

```diff
@@ -0,0 +1,96 @@
+import fs from 'fs/promises'
+import { existsSync } from 'fs'
+import type { LanguageRegistration, ThemeRegistration } from 'shiki/core'
+import type { ModuleOptions } from '../types'
+
+export async function mdcHighlighter({
+  options: {
+    shikiPath,
+    options
+  }
+}: {
+  options: {
+    shikiPath: string,
+    options: ModuleOptions['highlight']
+  }
+}) {
+  if (!options || !options.highlighter)
+    return 'export default () => { throw new Error(\'[@nuxtjs/mdc] No highlighter specified\') }'
+
+  if (options.highlighter === 'shiki') {
+    const file = [
+      shikiPath,
+      shikiPath + '.mjs',
+    ].find((file) => existsSync(file))
+
+    if (!file)
+      throw new Error(`[@nuxtjs/mdc] Could not find shiki highlighter: ${shikiPath}`)
+
+    const code = await fs.readFile(file, 'utf-8')
+
+    const { bundledLanguagesInfo } = await import('shiki/langs')
+
+    const langs = new Set<string | LanguageRegistration>()
+    options.langs?.forEach((lang) => {
+      if (typeof lang === 'string') {
+        const id = bundledLanguagesInfo.find(i => i.aliases?.includes?.(lang))?.id || lang
+        if (!bundledLanguagesInfo.find(i => i.id === id)) {
+          console.error(`[@nuxtjs/mdc] Could not find shiki language: ${lang}`)
+          return
+        }
+        langs.add(id)
+      }
+      else
+        langs.add(lang)
+    })
+
+    const themes = Array.from(new Set([
+      ...typeof options?.theme === 'string'
+        ? [options?.theme]
+        : Object.values(options?.theme || {}),
+      ...options?.themes || []
+    ]))
+
+    return [
+      'import { getMdcConfigs } from \'#mdc-configs\'',
+      code,
+
+      'const langs = [',
+      ...Array.from(langs)
+        .map((lang) => typeof lang === 'string'
+          ? `  import('shiki/langs/${lang}.mjs'),`
+          : '  ' + JSON.stringify(lang) + ','),
+      ']',
+      'const themes = [',
+      ...themes.map((theme: string | ThemeRegistration) => typeof theme === 'string'
+        ? `  import('shiki/themes/${theme}.mjs'),`
+        : '  ' + JSON.stringify(theme) + ','),
+      ']',
```

Comment:
This is great for built time, but I love to have a simple way to load themes and langs dynamically
Currently, I had to this in order to access shiki core and load themes based on highlight request
```ts
let shikiCore = null
  let setupThemes = null

  const shikiHighlighter = createShikiHighlighter({
    options: {},
    getMdcConfigs: () => {
      return [
        {
          shiki: {
            async setup (core) {
              shikiCore = core
              await importThemes(core, setupThemes)
            }
          }
        }
      ]
    }
  })
  const importThemes = async (core, theme) => {
    if (!theme) {
      return
    }
    if (typeof theme === 'string') {
      theme = { default: theme }
    }
    const loadedThemes = core.getLoadedThemes()
    for await (const themeName of Object.values(theme)) {
      if (!loadedThemes.includes(themeName)) {
        await core.loadTheme(import(`shiki/themes/${themeName}.mjs`).then(m => m.default || m))
      }
    }
  }
  const codeHighlighter = async (code, lang, theme, options) => {
    setupThemes = theme
    if (shikiCore) {
      await importThemes(shikiCore, setupThemes)
    }
    return shikiHighlighter(code, lang, theme, options)
  }
```